libs/sql-schema-describer/src/mssql.rs

```diff
@@ -0,0 +1,570 @@
+                AND t.is_ms_shipped = 0
+        "#;
+
+        debug!("describing indices, SQL: {}", sql);
+
+        let rows = self
+            .conn
+            .query_raw(sql, &[schema.into()])
+            .await
+            .expect("querying for indices");
+
+        for row in rows {
+            debug!("Got index row: {:#?}", row);
+
+            let table_name = row.get("table_name").and_then(|x| x.to_string()).expect("table_name");
+            let index_name = row.get("index_name").and_then(|x| x.to_string()).expect("index_name");
+
+            match row.get("column_name").and_then(|x| x.to_string()) {
+                Some(column_name) => {
+                    let seq_in_index = row.get("seq_in_index").and_then(|x| x.as_i64()).expect("seq_in_index");
+                    let pos = seq_in_index - 1;
+                    let is_unique = row.get("is_unique").and_then(|x| x.as_bool()).expect("is_unique");
+
+                    // Multi-column indices will return more than one row (with different column_name values).
+                    // We cannot assume that one row corresponds to one index.
+                    let (ref mut indexes_map, ref mut primary_key): &mut (_, Option<PrimaryKey>) = map
+                        .entry(table_name)
+                        .or_insert((BTreeMap::<String, Index>::new(), None));
+
+                    let is_pk = row
+                        .get("is_primary_key")
+                        .and_then(|x| x.as_bool())
+                        .expect("is_primary_key");
+
+                    if is_pk {
+                        debug!("Column '{}' is part of the primary key", column_name);
+
+                        match primary_key {
+                            Some(pk) => {
+                                if pk.columns.len() < (pos + 1) as usize {
+                                    pk.columns.resize((pos + 1) as usize, "".to_string());
+                                }
+
+                                pk.columns[pos as usize] = column_name;
+
+                                debug!(
+                                    "The primary key has already been created, added column to it: {:?}",
+                                    pk.columns
+                                );
+                            }
+                            None => {
+                                debug!("Instantiating primary key");
+
+                                primary_key.replace(PrimaryKey {
+                                    columns: vec![column_name],
+                                    sequence: None,
+                                    constraint_name: None,
+                                });
+                            }
+                        };
+                    } else if indexes_map.contains_key(&index_name) {
+                        if let Some(index) = indexes_map.get_mut(&index_name) {
+                            index.columns.push(column_name);
```

Comment:
are we sure the index columns are going to be in the right order? I don't see an `ORDER BY` clause in the query.

Reply:
Yeah. The ordering should be in the query. Fixing.